### PR TITLE
ISTE 173 - BUILD even with Develop

### PR DESCRIPTION
**Pull request by Hari-egov**

### CODE REVIEW

File: frontend/mgramseva/lib/components/house_connection_and_bill/new_consumer_bill.dart [hunk 1]
```diff
@@ -162,86 +155,208 @@ class NewConsumerBillState extends State<NewConsumerBill> {
...
+                                              .getPenaltyOrAdvanceStatus(
+                                                  widget.waterConnection
+                                                      ?.mdmsData,
+                                                  true) &&
+                                          houseHoldProvider.isfirstdemand)
+                                        getLabelText(
+                                            i18.common.CORE_NET_DUE_AMOUNT,
+                                            "${netDueAmount(houseHoldProvider)}",
+                                            context),
+                                      // if (CommonProvider
```
Comment (naveen-egov) [inline frontend/mgramseva/lib/components/house_connection_and_bill/new_consumer_bill.dart:265]: Remove un-used code

File: frontend/mgramseva/lib/providers/bill_generation_details_provider.dart [hunk 2]
```diff
@@ -134,8 +134,7 @@ class BillGenerationProvider with ChangeNotifier {
       var reqDate = readDate.add(Duration(days: 1)).toLocal().toString();
       prevReadingDate = DateFormats.dateToTimeStamp(
           DateFormats.getFilteredDate(reqDate, dateFormat: 'dd/MM/yyyy'));
```
Comment (naveen-egov) [inline frontend/mgramseva/lib/providers/bill_generation_details_provider.dart:136]: Create a date formate constant class and read from it.

File: frontend/mgramseva/lib/providers/collect_payment_provider.dart [hunk 3]
```diff
@@ -398,7 +398,7 @@ class CollectPaymentProvider with ChangeNotifier {
                                   .toString()) +
                               ' only')),
```
Comment (naveen-egov) [inline frontend/mgramseva/lib/providers/collect_payment_provider.dart:399]: use 18 constants

File: frontend/mgramseva/lib/providers/common_provider.dart [hunk 4]
```diff
@@ -79,7 +81,7 @@ class CommonProvider with ChangeNotifier {
       var response = await CoreRepository().getLocilisation(query);
       labels = localizedStrings = response;
       setLocalizationLabels(response);
-        } catch (e) {
+    } catch (e) {
```
Comment (naveen-egov) [inline frontend/mgramseva/lib/providers/common_provider.dart:84]: Remove Print statement

File: frontend/mgramseva/lib/providers/household_register_provider.dart [hunk 5]
```diff
@@ -183,34 +183,66 @@ class HouseholdRegisterProvider with ChangeNotifier {
...
-  ;
+        TableHeader(i18.common.CONNECTION_ID,
+            isSortingRequired: true,
+            isAscendingOrder:
+                sortBy != null && sortBy!.key == 'connectionNumber'
+                    ? sortBy!.isAscending
+                    : null,
+            apiKey: 'connectionNumber',
+            callBack: onSort),
+        // TableHeader(i18.common.VILLAGE_CODE, isSortingRequired: false),
```
Comment (naveen-egov) [inline frontend/mgramseva/lib/providers/household_register_provider.dart:194]: Remove un-used code

File: utilities/Penalty/src/main/resources/application.properties [hunk 6]
```diff
@@ -10,7 +10,7 @@ mdms.event.tenantId=pb
 egov.wscalculator.host=http://localhost:8083
 egov.wscalculator.url=/ws-calculator/waterCalculator/_addPenalty
 egov.penalty.enabled= true
-egov.penalty.enabled.division.code.list=DIV1038,DIV1001
+egov.penalty.enabled.division.code.list=DIV1038
```
Comment (naveen-egov) [inline utilities/Penalty/src/main/resources/application.properties:13]: What is the reason this change

**Review by naveen-egov — CHANGES_REQUESTED** [review]

File: frontend/mgramseva/lib/providers/bill_generation_details_provider.dart (hunk 2 above)
Comment (Hari-egov) [reply frontend/mgramseva/lib/providers/bill_generation_details_provider.dart:136]: DateFormats is already a static class,
which is accessed from egov\mGramseva-HRMS\punjab-mgramseva\frontend\mgramseva\lib\utils\date_formats.dart

File: frontend/mgramseva/lib/providers/collect_payment_provider.dart (hunk 3 above)
Comment (Hari-egov) [reply frontend/mgramseva/lib/providers/collect_payment_provider.dart:399]: English Only Area,
where number to words translation

File: utilities/Penalty/src/main/resources/application.properties (hunk 6 above)
Comment (Hari-egov) [reply utilities/Penalty/src/main/resources/application.properties:13]: Back end change,

File: frontend/mgramseva/lib/providers/household_register_provider.dart (hunk 5 above)
Comment (Hari-egov) [reply frontend/mgramseva/lib/providers/household_register_provider.dart]: Done

File: frontend/mgramseva/lib/providers/common_provider.dart (hunk 4 above)
Comment (Hari-egov) [reply frontend/mgramseva/lib/providers/common_provider.dart]: Done

File: frontend/mgramseva/lib/components/house_connection_and_bill/new_consumer_bill.dart (hunk 1 above)
Comment (Hari-egov) [reply frontend/mgramseva/lib/components/house_connection_and_bill/new_consumer_bill.dart]: Done

**Review by pradeepkumarcm-egov — APPROVED** [review]

**Pull request merged**